### PR TITLE
Reduce average query size sent to Solr

### PR DESCRIPTION
**Pull request by pklinef**
## What does this PR do?
Reduces AnyText query parameters from 3200 characters down to 200 characters.  Now by default, only the metadata, title, description fields will be queried for AnyText queries.  This can be configured with the `solr.query.anytext.fields` system property to add more fields based on user needs.

- Removes the dynamic generation of AnyText fields. This creates poor
  query performance when expanding to hundreds of fields.
- Removes filtered attribute feature that generated large field lists
  but failed to significantly reduce query time for majority of queries.
- Removes spellcheck parameter if spell check is not enabled for the query.
- Removes score from field list if not sorting by relevance.

#### Select relevant component teams: 

@codice/solr 

#### How should this be tested?

Ingest data and perform AnyText queries for tokens in the metadata, title, description fields.

#### Any background context you want to provide?
https://www.slideshare.net/lucidworks/query-hundreds-of-fields-at-scale-yannis-hector-salesforce-david-smiley-d-w-smiley-llc?qid=96050cd7-1b2b-4ee9-97c3-8dab60a66741&v=&b=&from_search=1

https://www.youtube.com/watch?v=hqeYAnsxPH8

#### What are the relevant tickets?
Fixes: n/a

#### Screenshots

Before
```
solr1    | 2020-09-17 18:42:18.879 INFO  (qtp631673932-724) [c:catalog s:shard5 r:core_node21 x:catalog_shard5_replica_n18] o.a.s.c.S.Request [catalog_shard5_replica_n18]  webapp=/solr path=/select params={q=+(+(point-of-contact_txt_tokenized:"test"+source-id_txt_tokenized:"test"+metacard.deleted.tags_txt_tokenized:"test"+contact.publisher-name_txt_tokenized:"test"+metacard.deleted.id_txt_tokenized:"test"+validation-errors_txt_tokenized:"test"+media.type_txt_tokenized:"test"+id_txt_tokenized:"test"+ext.extracted.text_txt_tokenized:"test"+contact.creator-email_txt_tokenized:"test"+contact.creator-address_txt_tokenized:"test"+metacard_type_name_txt_tokenized:"test"+checksum_txt_tokenized:"test"+metacard.version.edited-by_txt_tokenized:"test"+metacard.deleted.version_txt_tokenized:"test"+checksum-algorithm_txt_tokenized:"test"+metacard.version.resource-uri_txt_tokenized:"test"+metadata-content-type-version_txt_tokenized:"test"+security.access-groups_txt_tokenized:"test"+title_txt_tokenized:"test"+validation-warnings_txt_tokenized:"test"+media.format_txt_tokenized:"test"+metacard.owner_txt_tokenized:"test"+contact.publisher-address_txt_tokenized:"test"+contact.creator-phone_txt_tokenized:"test"+security.access-individuals_txt_tokenized:"test"+topic.keyword_txt_tokenized:"test"+contact.publisher-email_txt_tokenized:"test"+media.encoding_txt_tokenized:"test"+topic.category_txt_tokenized:"test"+failed-validators-warnings_txt_tokenized:"test"+resource.derived-download-url_txt_tokenized:"test"+metacard.associations.derived_txt_tokenized:"test"+contact.contributor-email_txt_tokenized:"test"+contact.point-of-contact-name_txt_tokenized:"test"+failed-validators-errors_txt_tokenized:"test"+metacard.version.type_txt_tokenized:"test"+resource-size_txt_tokenized:"test"+security.access-administrators_txt_tokenized:"test"+media.compression_txt_tokenized:"test"+metacard.associations.related_txt_tokenized:"test"+location.country-code_txt_tokenized:"test"+metadata-content-type_txt_tokenized:"test"+contact.creator-name_txt_tokenized:"test"+contact.contributor-phone_txt_tokenized:"test"+security.access-groups-read_txt_tokenized:"test"+contact.publisher-phone_txt_tokenized:"test"+metadata_txt_tokenized:"test"+contact.point-of-contact-email_txt_tokenized:"test"+topic.vocabulary_txt_tokenized:"test"+metacard.version.tags_txt_tokenized:"test"+contact.point-of-contact-address_txt_tokenized:"test"+metacard.version.action_txt_tokenized:"test"+language_txt_tokenized:"test"+metacard.version.id_txt_tokenized:"test"+datatype_txt_tokenized:"test"+media.scanning-mode_txt_tokenized:"test"+description_txt_tokenized:"test"+associations.external_txt_tokenized:"test"+datetime.name_txt_tokenized:"test"+security.access-individuals-read_txt_tokenized:"test"+metacard.deleted.deleted-by_txt_tokenized:"test"+metacard-tags_txt_tokenized:"test"+resource-uri_txt_tokenized:"test"+contact.contributor-name_txt_tokenized:"test"+contact.point-of-contact-phone_txt_tokenized:"test"+location.crs-name_txt_tokenized:"test"+resource.derived-uri_txt_tokenized:"test"+resource-download-url_txt_tokenized:"test"+contact.contributor-address_txt_tokenized:"test"+metadata-target-namespace_txt_tokenized:"test"+media.format-version_txt_tokenized:"test"+location.crs-code_txt_tokenized:"test")+AND+metacard-tags_txt_tokenized:"resource"+)+&_stateVer_=catalog:9&spellcheck=false&start=0&rows=1&wt=javabin&version=2} hits=1268 status=0 QTime=64
```

After
```
solr1    | 2020-09-17 18:06:46.004 INFO  (qtp631673932-132) [c:catalog s:shard6 r:core_node24 x:catalog_shard6_replica_n22] o.a.s.c.S.Request [catalog_shard6_replica_n22]  webapp=/solr path=/select params={q=+(+(title_txt_tokenized:"test"+description_txt_tokenized:"test"+metadata_txt_tokenized:"test")+AND+metacard-tags_txt_tokenized:"resource"+)+&_stateVer_=catalog:9&start=0&rows=1&wt=javabin&version=2} hits=1268 status=0 QTime=19
```

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [x] Update / Add Unit Tests
- [x] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by pklinef** [comment]
build now

File: distribution/ddf-common/src/main/resources-filtered/etc/custom.system.properties [hunk 1]
```diff
@@ -87,7 +87,7 @@ solr.mem=2g
 solr.useBasicAuth=true
 solr.username=admin
 
-
+solr.query.anytext.fields=metadata,title,description
```
Comment (brendan-hofmann) [inline distribution/ddf-common/src/main/resources-filtered/etc/custom.system.properties:90]: I think we should include `ext.extracted-text` by default as well.

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java [hunk 2]
```diff
@@ -906,7 +834,24 @@ String getSortKey(String field) {
...
+            (PrivilegedAction<String>) () -> System.getProperty(SOLR_ANYTEXT_FIELDS, ""));
+
+    List<String> fields;
+    if (property.isEmpty()) {
+      fields = Arrays.asList(Metacard.METADATA, Metacard.TITLE, Metacard.DESCRIPTION);
+    } else {
+      fields = Arrays.asList(property.split(","));
+    }
+
+    return fields.stream()
```
Comment (Lambeaux) [inline catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java:852]: ❓  Should we filter blank strings to account for redundant commas in the system property?

File: catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/provider/SolrProviderQuery.java [hunk 3]
```diff
@@ -766,37 +763,6 @@ public void testContextualAnyText() throws Exception {
         1, getFilterBuilder().attribute(Metacard.ANY_TEXT).is().like().text(soughtWord), provider);
   }
 
-  @Test
-  public void testQueryExcludedAttributes() throws Exception {
```
Comment (Lambeaux) [inline catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/provider/SolrProviderQuery.java:770]: ❓  Not sure how meaningful the implications are of not sending the spell check option to Solr. Worth writing a test for it?

**Review by Lambeaux — APPROVED** [review]

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java (hunk 2 above)
Comment (Lambeaux) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java:852]: ❓ Would it be worth adding a test for that case as well?

File: catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/provider/SolrProviderQuery.java (hunk 3 above)
Comment (pklinef) [reply catalog/solr/catalog-solr-core/src/test/java/ddf/catalog/source/solr/provider/SolrProviderQuery.java:770]: Not sending it is the default.  It is what we were doing before the spell check PR.

File: catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java (hunk 2 above)
Comment (pklinef) [reply catalog/solr/catalog-solr-core/src/main/java/ddf/catalog/source/solr/DynamicSchemaResolver.java:852]: Probably not a bad idea.  I can also look at removing whitespace padding too if it is present.

**Review by jrnorth — APPROVED** [review]

**Review by derekwilhelm — APPROVED** [review]

**Comment by pklinef** [comment]
build now

**Pull request merged**